src/internal/operators/switchScan.ts

```diff
@@ -0,0 +1,63 @@
+import { Observable } from '../Observable';
+import { ObservableInput, OperatorFunction, MonoTypeOperatorFunction } from '../types';
+import { Operator } from '../Operator';
+import { Subscriber } from '../Subscriber';
+import { of } from '../observable/of';
+import { switchMap } from './switchMap';
+import { tap } from './tap';
+
+export function switchScan<T>(accumulator: (acc: T, value: T, index: number) => ObservableInput<T>, seed?: T): MonoTypeOperatorFunction<T>;
+export function switchScan<T, R>(accumulator: (acc: R, value: T, index: number) => ObservableInput<R>, seed?: R): OperatorFunction<T, R>;
+
+/**
+ * Applies an accumulator function over the source Observable where the
+ * accumulator function itself returns an Observable, emitting values
+ * only from the most recently returned Observable.
+ *
+ * <span class="informal">It's like {@link scan}, but only the most recent
+ * Observable returned by the accumulator is merged into the outer Observable.</span>
+ *
+ * ![](switchScan.png)
+ *
+ * @see {@link scan}
+ * @see {@link mergeScan}
+ *
+ * @param {function(acc: R, value: T, index: number): Observable<R>} accumulator
+ * The accumulator function called on each source value.
+ * @param {T|R} [seed] The initial accumulation value.
+ * @return {Observable<R>} An observable of the accumulated values.
+ * @method switchScan
+ * @owner Observable
+ */
+export function switchScan<T, R>(
+  accumulator: (acc: R, value: T, index: number) => ObservableInput<R>,
+  seed?: R
+): OperatorFunction<T, R> {
+  let hasSeed = false;
+  // The same reason as described in `scan` https://github.com/ReactiveX/rxjs/blob/master/src/internal/operators/scan.ts#L54-L58
+  if (arguments.length >= 2) {
+    hasSeed = true;
+  }
+
+  return (source: Observable<T>) => source.lift(new SwitchScanOperator(accumulator, seed, hasSeed));
+}
+
+class SwitchScanOperator<T, R> implements Operator<T, R> {
+  constructor(private accumulator: (acc: R, value: T, index: number) => ObservableInput<R>, private seed: R, private hasSeed: boolean = false) { }
```

Comment:
The seed is mutated, so it cannot be stored in a property that belongs to the operator instance. The operator instance forms part of the composed observable chain. Stored here, the seed will not be per-subscriber, so the subscribers will interfere with one another. The `source.pipe` call could be made within a `defer` call and the seed could be stored within the callback passed to `defer`.

Reply:
Yeah, you're right. It really was overriding the same seed among multiple subscribers. I think I don't even need `defer` and just use scoped variables. I fixed it and added a test that would fail otherwise.

Reply:
Yeah, `call` is called per-subscriber, so shoving it in there as a variable is fine.